mysys_ssl/my_aes_openssl.cc

```diff
@@ -108,33 +108,35 @@ int my_aes_encrypt(const unsigned char *source, uint32 source_length,
                    const unsigned char *key, uint32 key_length,
                    enum my_aes_opmode mode, const unsigned char *iv)
 {
-  EVP_CIPHER_CTX ctx;
+  EVP_CIPHER_CTX *ctx = EVP_CIPHER_CTX_new();
+  if (unlikely(!ctx))
+    return MY_AES_BAD_DATA;
   const EVP_CIPHER *cipher= aes_evp_type(mode);
   int u_len, f_len;
   /* The real key to be used for encryption */
   unsigned char rkey[MAX_AES_KEY_LENGTH / 8];
   my_aes_create_key(key, key_length, rkey, mode);
 
   if (!cipher || (EVP_CIPHER_iv_length(cipher) > 0 && !iv))
-    return MY_AES_BAD_DATA;
+    goto aes_error;
```

Comment:
This jump leads to additional `ERR_clear_error` call, I'm not sure if we lose any useful error information after that, upstream just does here
```
EVP_CIPHER_CTX_free(ctx);
return MY_AES_BAD_DATA;
```

Reply:
I think either path is fine. Since this condition isn't triggered by an OpenSSL error, there won't be an error there to clear. So clearing it anyway is vacuous. I did this because it avoids duplication of the error handling. I suppose we should just use upstream, though.